tests/third_party/cupy/math_tests/test_arithmetic.py

```diff
@@ -99,13 +129,36 @@ def test_raises_with_numpy_input(self):
     )
 )
 @pytest.mark.usefixtures("allow_fall_back_on_numpy")
-class TestArithmeticUnary(unittest.TestCase):
+class TestArithmeticUnary:
     @testing.numpy_cupy_allclose(atol=1e-5, type_check=has_support_aspect64())
     def test_unary(self, xp):
         arg1 = self.arg1
         if isinstance(arg1, numpy.ndarray):
             arg1 = xp.asarray(arg1)
-        y = getattr(xp, self.name)(arg1)
+
+        if self.name in ("reciprocal") and xp is numpy:
+            # In Numpy, for integer arguments with absolute value larger than 1 the result is always zero.
```

Comment:
I believe we simply replicate the behavior of `1/x` which is indicated by NumPy and cupy as well.
@ndgrigorian is this the reason?

Reply:
Imitating the behavior of division was chosen over imitating Numpy's current reciprocal behavior intentionally. There is some discussion by Numpy about how the function is outdated (apparently closer to Python 2.x's division behavior) among other things.

As a result, we opted to break with Numpy.

Reply:
@ndgrigorian , thank you! it's clear now.

@vtavana, probably it would be good to leave some note in the description of `dpnp.reciprocal` to clear describe the behavior with an array of integer dtype, since the current logic differs from numpy?